### PR TITLE
Change backend.CurrentUser to return service accounts

### PR DESCRIPTION
**Pull request by Frassle**
# Description

Change `about` and `whoami` to return the username as "organization" when using the filestate backend, this matches what's expected for their fully qualified stack references. 

As part of this we change the internal backend interface such that `CurrentUser` is service specific returning a full `workspace.Account` (see https://github.com/pulumi/pulumi/pull/13206#discussion_r1258716013 for some context on why we want to do this).

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: pkg/backend/httpstate/backend.go [hunk 1]
```diff
@@ -480,33 +480,41 @@ func (b *cloudBackend) Name() string {
...
 }
 
-func (b *cloudBackend) CurrentUser() (string, []string, error) {
-	return b.currentUser(context.Background())
+func (b *cloudBackend) CurrentUser() (*workspace.Account, error) {
+	account, err := b.currentUser(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	return &account, nil
```
Comment (abhinav) [inline pkg/backend/httpstate/backend.go:499]: optional: can shorten this to:

```go
return &account, err
```

It's understood that the returned value is a undefined if the error is non-nil.

File: pkg/backend/httpstate/backend.go [hunk 2]
```diff
@@ -526,11 +534,11 @@ func (b *cloudBackend) parsePolicyPackReference(s string) (backend.PolicyPackRef
 	}
 
 	if orgName == "" {
-		currentUser, _, userErr := b.CurrentUser()
+		currentUser, userErr := b.currentUser(context.Background())
```
Comment (abhinav) [inline pkg/backend/httpstate/backend.go:537]: optional nit: Any reason why this isn't just 'err' instead of 'userErr'?
(I understand that this is prior code so feel free to defer anything actionable here if you'd prefer.)

Comment (Frassle) [reply pkg/backend/httpstate/backend.go:537]: Yeh this can just be err.

File: pkg/backend/httpstate/backend.go [hunk 3]
```diff
@@ -480,33 +480,41 @@ func (b *cloudBackend) Name() string {
...
+
 	if account.Username != "" {
-		logging.V(1).Infof("found username for access token")
-		return account.Username, account.Organizations, nil
+		logging.V(1).Infof("no username for access token")
+		name, orgs, err := b.client.GetPulumiAccountDetails(ctx)
+		if err != nil {
+			return workspace.Account{}, err
+		}
+		account.Username = name
```
[the remark this answers is not in the document]
Comment (Frassle) [reply pkg/backend/httpstate/backend.go:514]: Good spot

File: pkg/backend/httpstate/backend.go (hunk 1 above)
Comment (Frassle) [reply pkg/backend/httpstate/backend.go:499]: Sure

File: pkg/backend/httpstate/backend.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (abhinav) [reply pkg/backend/httpstate/backend.go]: Sorry, clarify? Are we not saving the account information here?

Also, just looking at the implementation in here, I'm not sure we should use GetAccount here? That'll load the credentials from disk again which (besides being a leak of information from the workspace into a backend which we may or may not care about right now) was probably already done once before by the time we got here in the backend (because we probably had to construct a client).
I may be missing something but it looks like we can just unconditionally call GetPulumiAccountDetails—it'll record the user information it gets back from the server.

[the remark this answers is not in the document]
Comment (Frassle) [reply pkg/backend/httpstate/backend.go]: >Sorry, clarify? Are we not saving the account information here? 

No we'd need a StoreAccount call, which we do on login or credential refresh.

But yeh I'll look into this a bit more, I suspect you might be right that we'll already have an account from when we set the backend up.
